ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/TransactionServiceIT.scala

```diff
@@ -1582,26 +1582,32 @@ object TransactionServiceIT {
 
   // Strip command id and offset to yield a transaction comparable across participant
   // Furthermore, makes sure that the order is not relevant for witness parties
+  // Sort by transactionId as on distributed ledgers updates can occur in different orders
   private def comparableTransactions(transactions: Vector[Transaction]): Vector[Transaction] =
-    transactions.map(t =>
-      t.copy(
-        commandId = "commandId",
-        offset = "offset",
-        events = t.events.map(_.modifyWitnessParties(_.sorted)),
+    transactions
+      .map(t =>
+        t.copy(
+          commandId = "commandId",
+          offset = "offset",
+          events = t.events.map(_.modifyWitnessParties(_.sorted)),
+        )
       )
-    )
+      .sortBy(_.transactionId)
```

Comment:
I _think_ these sorting changes are fine, but I'm not personally aware of how the transaction service is used in practice. It might be that the transactions are expected to be sorted by applications and, for example, triggers, in which case we should probably be making the ledger do it.

Any idea if anyone is reliant on transaction sort order?

Reply:
The sorting is fine. If 2 contracts don't have a relation to each other, the order in which they appear on the transaction stream doesn't matter. You could view this as partial ordering.
This ordering change should only be relevant for the test case `TXSingleMultiSameStakeholders` where the a distributed ledger might re-order the transactions that are created on alpha and beta for a specific party.

Reply:
Awesome, thanks for the clarification!

Reply:
fwiw we do have docs on causality which also include ordering guarantees we make or explicitly don’t make https://docs.daml.com/concepts/local-ledger.html

Reply:
Yes there are ordering guarantees (like the create must come before an exercise).  The tests using these methods are not concerned with ensuring ordering guarantees (they only create contracts with no contractId arguments).